### ISSUE TITLE
Ограничение на полное имя файла базы данных в MON$ATTACHMENTS

### ISSUE THREAD

**Issue by pincher1519** [FirebirdSQL/firebird#7240]
Приложение проверяет количество подключений к базе разных других пользователей, с других компьютеров и так далее.
Баз данных множество (достигать может не то что бы сотен, но и тысяч файлов) и пользоваться aliases не удобно.
Базы разложены по каталогам и иногда их путь превышает ограничение указанное в **MON$ATTACHMENTS.MON$ATTACHMENT_NAME**

При попытке получить количество подключений к базе, а также информацию о том, кто подключен к базе - выдается ошибка. Данную таблицу невозможно открыть и при помощи IBExpert
![Безымянный](https://user-images.githubusercontent.com/34896014/180797816-7a46958a-b4c9-4327-9d52-06522b678658.png)

Использовался **Firebird_3-0-8-33560**, но в более свежих такая же проблема.
Изменить системную таблицу не получается.

При подключении к базам используются UNC пути `"\\?\"`

Планируется ли разрешить получение данных о подключениях из системной таблицы для баз, путь которых превышает 255 символов?

**Comment by sim1984** [comment]
UNC should never be used. It is not safe to share the database folder.
If you meant WNET, then it's better not to use it either. It is considered obsolete and will be removed in Firebird 5.0.
Tickets should be made in English so that all developers can read them, and not just Russian-speaking ones.

**Comment by aafemt** [comment]
It is not network UNC. It is a hack to make old API work with long paths.

**Comment by pincher1519** [comment]
> UNC should never be used. It is not safe to share the database folder.

Yes, the system is designed in such a way that you have to work mainly with databases on Windows OS, for network versions of databases there is a separate service database described in aliases, which describes all local paths to databases that can be accessed.
There is no other way to figure out how to have access to thousands of bases from the outside to get their list. If you have an example of how a third-party utility can ask the server what databases it can connect and provide from a specific directory - I would love to hear it. But for now, everything is organized in such a way that clients connect bases using an absolute path on the server, recognizing their paths through a fixed base through an alias.

> If you meant WNET, then it's better not to use it either. It is considered obsolete and will be removed in Firebird 5.0. 

No, I don’t know about such functionality, I don’t use it

> Tickets should be made in English so that all developers can read them, and not just Russian-speaking ones.

Unfortunately, my English is very poor, so if you need to write, then it's through a translator

**Comment by pincher1519** [comment]
> It is not network UNC. It is a hack to make old API work with long paths.

One way or another, you need to connect databases along paths located up to 500+ characters with Russian-language directory names, which reduces the path size by half due to UTF 8. In total, with 255 bytes of data, ~128 characters remain for the full path to the file. This is not enough in modern realities.
The same Linux systems allow you to work with much larger paths (if I'm not mistaken, up to 32k characters), and in Windows you can access files using large paths by adding the UC prefix `"\\?\d:\Folder\folder\..."`.
There are other approaches how to connect to databases using long paths?

**Comment by aafemt** [comment]
In any case Firebird is using ANSI API for files. There is no way to use UTF-8 or expand length of path beyond 253 characters.

For thousands of databases I would suggest to use hexadecimal names derived from GUID. One short path can contain them all.
